### ISSUE TITLE
Midterm peer review - vap43

### ISSUE THREAD

**Issue by vineetparikh** [brian-liu-24/FeatureExplanationProject#4]
So this project is primarily about understanding the relationship between the accuracy of machine learning models and their corresponding explainability w.r.t. feature explanation. More specifically, two datasets (the Titanic dataset to predict whether a passenger survived the sinking of the Titanic, and the AMES Housing Price dataset to predict the price of a house in Boston) are considered, multiple models of varying complexity/explainability are fitted on these datasets (linear regression w/ridge regularization, XGBoost regression, etc.), and feature explanations are given based on models such as the SHAP explanation model.  The main goal is to see whether changing the accuracy of a predictive function (for instance, by changing the model type) will necessarily affect explainability of a model, and to see whether feature explanations are consistent for multiple applications on a single model. Future steps include including other popular feature explanation models (such as LIME), testing on more datasets, and finding out why feature explanation models perform more consistently for models with lower accuracy.

I definitely like that empirical tests have been added and run on datasets with multiple different schemas, that there is a solid algorithm for experimenting with altering accuracy (even if it's primarily based on taking different fractions of the training data and training on those fractions), and that there are solid metrics for understanding how explanable a given model would be (both via explanability models and through simple linear algebra e.g. the Kendall Tau distance between ranks).

However, it would be nice to see if there are more theoretical proofs for accuracy vs explanability, if there is a consideration of newer methods such as GA^2M (especially as this was recently lectured about and is advertised as a method to preserve a relatively high amount of accuracy while still possessing reasonable "explanability"), and if more model types (such as neural networks) could be considered when attempting to truly model the accuracy-explanability tradeoff.
